src/main/scala/strawman/collection/mutable/ArrayBuffer.scala

```diff
@@ -39,7 +39,7 @@ class ArrayBuffer[A] private (initElems: Array[AnyRef], initLength: Int)
         start = 0
       }
       else {
-        val newelems = new Array[AnyRef](end * 2)
+        val newelems = new Array[AnyRef](if (end == 0) 16 else (end * 2))
```

Comment:
The overflow cuts the usable size for arrays in half. Once you exceed MaxInt/2 size any attempt to grow the buffer fails even though you should be able to grow up to MaxInt size.